File: src/main/java/org/zwobble/mammoth/styles/parsing/StyleMapParser.java
package org.zwobble.mammoth.styles.parsing;

import org.parboiled.support.Var;
import org.zwobble.mammoth.styles.StyleMap;
import org.zwobble.mammoth.styles.StyleMapBuilder;

import java.util.List;

import static java.util.Arrays.asList;

public class StyleMapParser {
    public static StyleMap parse(String input) {
        return parse(asList(input.split("\\r?\\n")));
    }

    public static StyleMap parse(List<String> lines) {
        Var<StyleMapBuilder> styleMap = new Var<>(StyleMap.builder());
        for (String line : lines) {
            handleLine(styleMap, line);
        }
        return styleMap.get().build();
    }

    private static void handleLine(Var<StyleMapBuilder> styleMap, String line) {
        if (line.startsWith("#")) {
            return;
        }
        line = line.trim();
        if (line.isEmpty()) {
            return;
        }
        Parsing.parse(StyleMappingParser.class, parser -> parser.StyleMapping(styleMap), line);
    }
}

File: dotnet/Mammoth/Couscous/mammoth/internal/styles/parsing/StyleMapParser.cs
using Mammoth.Couscous.java.util;
using Mammoth.Couscous.org.zwobble.mammoth.@internal.styles;

namespace Mammoth.Couscous.org.zwobble.mammoth.@internal.styles.parsing {
    internal static class StyleMapParser {
        internal static StyleMap parseStyleMappings(List<string> mappings) {
            return DefaultStyleMap();
        }
        
        internal static StyleMap parse(string styleMap) {
            return DefaultStyleMap();
        }

        private static StyleMap DefaultStyleMap() {
            return StyleMap.builder()

                .mapParagraph(ParagraphMatcher.styleName("Normal"), HtmlPath.element("p"))

                .build();
        }
    }
}
